cmake/find/llvm.cmake

```diff
@@ -51,47 +35,88 @@ if (ENABLE_EMBEDDED_COMPILER)
             set (LLVM_FOUND 0)
             set (USE_EMBEDDED_COMPILER 0)
         endif ()
-
-        # TODO: fix llvm 8+ and remove:
-        if (LLVM_FOUND AND LLVM_VERSION_MAJOR GREATER 7)
-            message(WARNING "LLVM 8+ not supported yet, disabling.")
-            set (USE_EMBEDDED_COMPILER 0)
-        endif ()
     else()
-        set (LLVM_FOUND 1)
-        set (USE_EMBEDDED_COMPILER 1)
-        set (LLVM_VERSION "7.0.0bundled")
-        set (LLVM_INCLUDE_DIRS
-            ${ClickHouse_SOURCE_DIR}/contrib/llvm/llvm/include
-            ${ClickHouse_BINARY_DIR}/contrib/llvm/llvm/include
-            ${ClickHouse_SOURCE_DIR}/contrib/llvm/clang/include
-            ${ClickHouse_BINARY_DIR}/contrib/llvm/clang/include
-            ${ClickHouse_BINARY_DIR}/contrib/llvm/llvm/tools/clang/include
-            ${ClickHouse_SOURCE_DIR}/contrib/llvm/lld/include
-            ${ClickHouse_BINARY_DIR}/contrib/llvm/lld/include
-            ${ClickHouse_BINARY_DIR}/contrib/llvm/llvm/tools/lld/include)
-        set (LLVM_LIBRARY_DIRS ${ClickHouse_BINARY_DIR}/contrib/llvm/llvm)
+        if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_CURRENT_BINARY_DIR)
+            message(WARNING "Option ENABLE_EMBEDDED_COMPILER is set but LLVM library cannot build if build directory is the same as source directory.")
```

Comment:
Why it's not possible?

Reply:
LLVM build script prevents it just in case (to avoid pollution of source tree and avoid stale header files from previous build). If we remove this check from LLVM, it will start to work.